### DISCUSSION TITLE
Dual sink/output to both i2S and fft

### DISCUSSION THREAD

**Discussion by stanleyseow** [pschatzmann/ESP32-A2DP#359] [Q&A] [answered]
I build a fft audio visualizer to display the freq bins using this example
https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/basic-a2dp-fft/basic-a2dp-fft.ino

I also can output the BT audio to a pcm5102 dac decoder via i2S. ( bt_music_receiver_simple )

The question is , how can I have the music output to BOTH ffp and i2s cos now the audio visualizer , I can't hear any music...

Thanks

**Comment by pschatzmann** [comment, marked as answer]
There are 2 obvious solutions:
1) continue to use the i2s output of the A2DP library and just change the flag when you register the callback!
2) Send the output to a MultiOutput which conisists of an I2SStream and a AudioRealFFT

**Comment by stanleyseow** [comment]
Thanks for the quick reply...

>>continue to use the i2s output of the A2DP library and just change the flag when you register the callback!

Can u explain more on this part "just change the flag when you register the callback"

Do you have an example code for this ?

I found this callback example... 
https://github.com/pschatzmann/ESP32-A2DP/blob/main/examples/bt_music_sender_ssid_callback/bt_music_sender_ssid_callback.ino

**Reply by pschatzmann** [reply]
Search the Readme of the A2DP Library for set_stream_reader or Just look at the explanation in the generated class documentation.

You should not really ask for things which are properly documented!

**Comment by stanleyseow** [comment]
Thanks, issue solved ....
Reread the docs and had to turn it to **true** as the default is false..

a2dp_sink.set_stream_reader(writeDataStream, true);
